NMS-14632: Add alt text to Operation (pt. 1) (#5660)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/user-management/pre-authentication.adoc

= Web UI Pre-Authentication

You can configure your {page-component-title} instance to run behind a proxy that provides authentication.
Authenticated users can then be directed to the {page-component-title} web UI using a header.
You can configure pre-authentication settings in `$\{OPENNMS_HOME}/jetty-webapps/opennms/WEB-INF/spring-security.d/header-preauth.xml`.
This file is automatically included in the Spring security context, but is not enabled by default.

> WARNING: Do not configure {page-component-title} to use pre-authentication unless you are certain that the web UI is accessibly only to the proxy, and not to end users.

Reviewer: ```suggestion
WARNING: Do not configure {page-component-title} to use pre-authentication unless you are certain that the web UI is accessible only to the proxy, and not to end users.
```

If end users have access to the web UI, malicious attackers can craft queries that include the pre-authentication header, allowing them full control of the web UI and any REST APIs.

== Enable pre-authentication

Set the `enabled` property in `header-preauth.xml`:

[source, xml]
<beans:property name="enabled" value="true" />

== Configure pre-authentication

You can configure the following properties to change the pre-authentication plugin's behavior:

[cols="1,2,1"]
|====
| Property  | Description | Default Value

| enabled
| Indicates whether the pre-authentication plugin is active.
| false

| failOnError
| If `true`, block users from logging in if their header is not set or the user does not exist. +

> If `fales`, fall through to other mechanisms (basic authentication, form login, and so on).

Reviewer: ```suggestion
If `false`, fall through to other mechanisms (basic authentication, form login, and so on).
```

| false

| userHeader

> | The HTTP header that specifies which user is authenticated.

Reviewer: ```suggestion
| The HTTP header that specifies which user to authenticate.
```

</beans:bean>
----

> After the pre-authentication plugin is enabled and set up, you can test its configuration using the following console command:

Reviewer: ```suggestion
After the pre-authentication plugin is enabled and set up, use the following command to test its configuration:
```
